squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/impl/FrontierImpl.java

```diff
@@ -267,15 +278,25 @@ public void run() {
 
     @Override
     public void addNewUris(List<CrawleableUri> uris) {
+        domainUriGenerator = new DomainBasedUriGenerator();
+        wellKnownUriGenrator = new WellKnownPathUriGenerator();
```

Comment:
Creating the generators again and again is not a good idea. Apart from that, it would be better if the user could define which generators he would like to have using the Spring XML file.